firebase-auth/src/commonMain/kotlin/dev/gitlive/firebase/auth/auth.kt

```diff
@@ -18,42 +18,77 @@ expect fun Firebase.auth(app: FirebaseApp): FirebaseAuth
 expect class FirebaseAuth {
     val currentUser: FirebaseUser?
     val authStateChanged: Flow<FirebaseUser?>
-    suspend fun sendPasswordResetEmail(email: String)
-    suspend fun signInWithEmailAndPassword(email: String, password: String): AuthResult
+    val idTokenChanged: Flow<FirebaseUser?>
+    var languageCode: String
+    suspend fun applyActionCode(code: String)
+    suspend fun checkActionCode(code: String): ActionCodeResult
+    suspend fun confirmPasswordReset(code: String, newPassword: String)
     suspend fun createUserWithEmailAndPassword(email: String, password: String): AuthResult
+    suspend fun fetchSignInMethodsForEmail(email: String): SignInMethodQueryResult
+    suspend fun sendPasswordResetEmail(email: String, actionCodeSettings: ActionCodeSettings? = null)
+    suspend fun sendSignInLinkToEmail(email: String, actionCodeSettings: ActionCodeSettings)
+    suspend fun signInWithEmailAndPassword(email: String, password: String): AuthResult
     suspend fun signInWithCustomToken(token: String): AuthResult
     suspend fun signInAnonymously(): AuthResult
     suspend fun signInWithCredential(authCredential: AuthCredential): AuthResult
     suspend fun signOut()
+    suspend fun updateCurrentUser(user: FirebaseUser)
+    suspend fun verifyPasswordResetCode(code: String): String
 }
 
 expect class AuthResult {
     val user: FirebaseUser?
 }
 
-expect class FirebaseUser {
-    val uid: String
-    val displayName: String?
-    val email: String?
-    val phoneNumber: String?
-    val isAnonymous: Boolean
-    suspend fun delete()
-    suspend fun reload()
-    suspend fun sendEmailVerification()
+expect class ActionCodeResult {
+    val operation: Operation
+    fun <T, A: ActionCodeDataType<T>> getData(type: A): T?
+}
+
+expect class SignInMethodQueryResult {
+    val signInMethods: List<String>
+}
+
+enum class Operation {
+    PasswordReset,
+    VerifyEmail,
+    RecoverEmail,
+    Error,
+    SignInWithEmailLink,
+    VerifyBeforeChangeEmail,
+    RevertSecondFactorAddition
+}
+
+sealed class ActionCodeDataType<T> {
+    object Email : ActionCodeDataType<String>()
+    object PreviousEmail : ActionCodeDataType<String>()
+    object MultiFactor : ActionCodeDataType<MultiFactorInfo>()
+}
+
+expect class ActionCodeSettings {
+    class Builder {
+        fun setAndroidPackageName(androidPackageName: String, installIfNotAvailable: Boolean, minimumVersion: String?): Builder
+        fun setDynamicLinkDomain(dynamicLinkDomain: String): Builder
+        fun setHandleCodeInApp(canHandleCodeInApp: Boolean): Builder
+        fun setIOSBundleId(iOSBundleId: String): Builder
+        fun setUrl(url: String): Builder
+        fun build(): ActionCodeSettings
+    }
```

Comment:
To make this more kotlin-style I suggest replacing the builder pattern with a data class with optional parameters like we do with `dev.gitlive.firebase.FirebaseOptions` (this reminds me to add a section in the read me about that)